reporter/src/funTest/kotlin/TestData.kt

```diff
@@ -19,9 +19,16 @@
 
 package org.ossreviewtoolkit.reporter
 
+import java.net.URI
```

Comment:
> Prevent the generation of empty reports by enabling a specific advisor
report only if there is an advisor result with findings of this
specific type.

I'm not sure I agree to this rationale. How do you then, as a user, distinguish between the cases where a report was not generated because it was not enabled at all, or because there were no findings? In other words, I would image users to prefer a "written confirmation" that there were *no* findings.

Reply:
Valid point. I think, what we actually want is to generate reports depending on the advisors that have run. But unfortunately, this information is not available yet in an ORT result; an advisor only occurs if it has generated findings.

So I think I will change this for now to always generate the defects report if there are any advisor results. Do you agree?

Reply:
Changed accordingly.

Reply:
> So I think I will change this for now to always generate the defects report if there are any advisor results. Do you agree?

I'm not sure. Wouldn't it be confusing to get a defects report if no advisor capable of providing defects is configured? What would be the contents of that report then? "No defects found" or could we do "No defect advisor configured" (which IMO would be preferable)?

Reply:
Currently, the report says that no defects were found.

As I said, the ORT result does not contain the information which advisor was run. I think, it would make sense to change this, but this would be a bigger effort; so this could be a topic for a separate PR.